File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
class Solution {
    public int[] solution(int[] arr) {
        int minIdx = 0;
        int min = arr[0];
        
        for(int i = 1; i < arr.length; i++){
            if(arr[i] < min){
                min = arr[i];
                minIdx = i;
            }
        }
        
        int[] answer = new int[arr.length - 1];
        for (int i = 0, j = 0; i < arr.length; i++) {
            if (i != minIdx) answer[j++] = arr[i];
        }
        if(answer.length == 0) return new int[]{-1};
        return answer;
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    auto it = min_element(arr.begin(), arr.end());

    if(it != arr.end()) arr.erase(it);
    if (arr.empty()) return {-1};
    else return arr;
}